### ISSUE TITLE
Two variables with the same name declared when using a lot of for loops

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#2203]
### Expected

```js
for (var n = 0;;);
for (var n1 = 0;;)
for (var n11 = 0;;);
```

### Actual

```js
for (var n = 0;;);
for (var n1 = 0;;)
for (var n11 = 0;;);
```

### Steps To Reproduce
I can't seem to get it to compile on deck.net.
http://deck.net/0d475c972b28f35fd4a01c4c0d2dbfcd
```cs
public class Program
{
    public static void Main()
    {
          for (int n = 0;;);
          for (int n = 0;;)
              for (int n1 = 0;;);
    }
}
```

**Comment by geoffreymcgill** [comment]
Looks like an endless loop is happening on Deck when running the sample.

**Comment by Zaid-Ajaj** [comment]
Compiling this in either Deck or Visual Studio [v15.6]
```csharp
 public static void Main()
 {
     for (int n = 0;;);
     for (int n = 0;;)
     for (int n2 = 0;;);
 }
```
Produces
```js
$main: function () {
    for (var n = 0; ; ) {
        ;
    }
}
```

Isn't this just the expected behavior? Looks like Bridge compiler has detected that the second `for-loop` to be unreachable code and didn't translate the code from that point

**Comment by michaelcheers** [comment]
Well, try this:
```csharp
for (var n = 0;false;)
for (var n=0;false;)
for (var n2=0;false;);
```
as the c#.

**Comment by michaelcheers** [comment]
This is simpler. [New Deck](http://deck.net/0d475c972b28f35fd4a01c4c0d2dbfcd)

**Comment by geoffreymcgill** [comment]
Looks the second `n` is just getting a incremented numerical suffix applied in order to make it unique. No "look ahead" is happening to determine if the new `var` is unique.

I'm not sure `n` actually needs to be auto-incremented in this scenario. The `n` is being redefined and set to `0` in both loops, so likely doesn't need the `1` index applied. 

We'll have to investigate and discuss. We'll keep this thread updated with our progress.

**Comment by Zaid-Ajaj** [comment]
@michaelcheers Looks like the bridge compiler **unnecessarily** re-names the variable of the subsequent `for-loop`s which then collides with the variable names of inner `for-loop`

**Comment by diab0l** [comment]
@Zaid-Ajaj Renaming is still neccessary for other scenarios.
Without renaming at the same scope, the following modification to your example fails: [JSBIN](https://jsbin.com/jodixekaza/edit?js,console).
The expected output in the first example is 10, but the actual output is 0.
That is, because in Javascript, unlike C#, the for loops do not introduce a new scope.

The corresponding [deck](http://deck.net/328ad4dfdedb70a2b4693ef7675988da) works fine because the second n gets renamed, even though it lives in a different scope on the C# end.

Remember that scoping is a pita in Javascript.

**Comment by samuelGrahame** [comment]
Wow, I was just looking at this, Its because c# moves all variables declared to the top. (IL), Very Interesting.

Good Catch guys :)

**Comment by geoffreymcgill** [comment]
A pull request was submitted a few hours ago with a fix for this issue. Once reviewed and testing is complete, the fix will be pushed to **master** and released with Bridge 15.7.0.
